vtex/address-form
# CHK-1958: Remove validation restriction of rules options

#### What is the purpose of this pull request?

This PR updates the logic for validating an address to not restrict the values in the address object to the ones defined in the `options` of the country rules.

#### What problem is this solving?

This fixes an increasingly common scenario where the store would modify the orderForm via API, inserting an address thay _may_ not "comply" to the rules defined in `address-form`. From simple cases of mismatching the casing of the value (e.g. using "RIO DE JANEIRO" instead of "Rio de Janeiro"), to values that are not mapped yet, it would cause an undesirable experience for the final user where they would need to update their address once more when arriving at Checkout.

#### How should this be manually tested?

You can test using [this workspace](https://lucas--vtexgame1.myvtex.com/checkout/cart/add?sku=312&qty=1&seller=1&sc=2).

Since this PR only handles the validation portion of the changes, there isn't much to check in the above workspace to assert its working, but I also linked there the changes from #470 (which complement the fix as a whole). You can use the same steps from the linked PR to also test this one.

#### Screenshots or example usage

N/A

#### Types of changes
